adagrams/game.py

```diff
@@ -1,11 +1,155 @@
-    pass
+  '''
+  input: two parameters word and letter_bank. word represents an input from the user
+  letter_bank represents random letters drawn at the start of the game
+  output: returns True if user used letters to make the word from letter_bank; False if they did not
+  '''
 
-def score_word(word):
-    pass
+  word = word.upper()
+  for letter in word:    
+    letter_bank_count = letter_bank.count(letter)
+    word_letter_count = word.count(letter)
+
+    if not letter in letter_bank or word_letter_count > letter_bank_count:
+        return False
+  return True
 
+def score_word(word):
+    '''
+    input: word is user input
+    output: integer representing the number of points scored based on the word submitted
+    '''
+    word = word.upper()
+    score = 0
+    if len(word) >= 7:
+       score += 8
+    for letter in word:
+       score += LETTER_POOL_VALUES.get(letter)
+    return score
+    
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    '''
+    input: word list is all of the words submitted by user while continuing to play
+    output: tuple that displays highest scoring word and its score
+    '''
+    word_list_score_dict = {}
+    checked_values = set()
+    duplicates = set()
+
+    for word in word_list:
+      score = score_word(word)
+      word_list_score_dict[word] = score
+      highest_score = max(word_list_score_dict.values())
+
+    #this second loop is initiated in order to check if multiple words received the same score  
+    for word, score in word_list_score_dict.items():
+      if score not in checked_values:
+        checked_values.add(score)
+      else:
+        duplicates.add(score)
+        
+    #conditional statement used if duplicate scores are found to call the helper function tie_breaker
+    if duplicates:
+      tie_breaker_score = tie_breaker(word_list)
+      return tie_breaker_score
+    
+    return (max(word_list_score_dict, key=word_list_score_dict.get)), highest_score
+    
+
+def tie_breaker(word_list):
+  '''
+  helper function that is used in get_highest_word_score if there are duplicates found 
+  input: word list
+  output: tuple that displays highest scoring word and its score
+  '''
+  word_count_list = []
+  shortest_word = []
+  tie_breaker_result = {}
+
+#tie_breaker function needs to give preference to ten letter words, so this helper functions is called here
+  prefers_ten_result = tie_breaker_prefers_ten(word_list)
+  
+  for word in word_list:
+    word_count_list.append(len(word))
+    tie_breaker_result[word] = (len(word))
+  
+  word_count_list.sort()
+  shortest_word = word_count_list[0]
+
+  #conditional is added here to return the result of helper function
+  if prefers_ten_result:
+    return prefers_ten_result
+  else:   
+    for key, value in tie_breaker_result.items():
+      if len(key) == shortest_word:
+          return key, score_word(key)
+      
+
+def tie_breaker_prefers_ten(word_list):
```

Comment:
This function name and function purpose is still a little confusing to me. I don't think that the name, input description, or output description are as descriptive or specific as they need to be. From the code, it looks like this word is finding the first word with a length of 10 and returning a tuple of its value and score.

Could we try names like `get_ten_length_word` or `get_preferred_word`?